src/ui/Facet/Facet.ts

```diff
@@ -1528,6 +1529,36 @@ export class Facet extends Component {
     this.dependsOnManager = new DependsOnManager(facetInfo);
   }
 
+  private dependsOnUpdateParentDisplayValue() {
+    if (!this.options.dependsOn) {
+      return;
+    }
+
+    const masterFacetComponent = ComponentsTypes.getAllFacetsInstance(this.root).filter((cmp: Facet) => {
+      return cmp.options.id === this.options.dependsOn;
+    });
+
+    if (!masterFacetComponent.length) {
+      this.logger.warn(
+        `Unable to find a Facet with the id or field "${this.options.dependsOn}".`,
+        `The master facet values can't be updated.`
+      );
+      return;
+    }
+    if (masterFacetComponent.length > 1) {
+      this.logger.warn(
+        `Multiple facets with id "${this.options.dependsOn}" found.`,
+        `A given facet may only depend on a single other facet.`,
+        `Ensure that each facet in your search interface has a unique id.`,
+        `The master facet cannot be updated.`,
+        masterFacetComponent
+      );
+      return;
+    }
+    const masterFacet = masterFacetComponent[0] as Facet;
```

Comment:
will do, but will still  need to cast as facet since `getAllFacetsInstance ` return a `component` so I can't update `keepDisplayedValuesNextTime` of the master facet.
Should be fine this time since we will know for sure  that it is an instance of `facet`

Reply:
Then I think this would make more sense:
```
const masterFacetComponent = ComponentsTypes.getAllFacetsInstance(this.root).filter((cmp: Component) => {
    const isFacet = cmp instanceof Facet;
    return isFacet && cmp.options.id === this.options.dependsOn;
}) as Facet[];
```
also after you could remove the casting on line 1559